src/main/scala/com/gu/invoicing/pdf/Program.scala

```diff
@@ -3,18 +3,40 @@ package com.gu.invoicing.pdf
 import com.gu.invoicing.pdf.Model._
 import com.gu.invoicing.pdf.Impl._
 import com.gu.invoicing.common.Retry._
+import com.gu.invoicing.common.ZuoraAuth.{GNMAustralia_InvoiceTemplateID}
+
+import java.lang.System.getenv
 
 object Program {
 
   /** Main business logic */
   def program(input: PdfInput): String = retryUnsafe {
     val PdfInput(invoiceId, identityId) = input
-    val Invoice(accountId, pdf) = getInvoice(invoiceId)
-    val actualIdentityId = getIdentityId(accountId)
+    val invoice = getInvoice(invoiceId)
+    assert(
+      invoiceId == invoice.Id,
+      s"Returned invoice id: ${invoice.Id} does not match requested invoice id: $invoiceId"
```

Comment:
Can this happen?

Reply:
Maybe before manage/MDAPI was aware of test users. But I think now it simply provides a weak second factor of authentication to the API.